test/browser/fragments.test.js

```diff
@@ -120,6 +130,79 @@ describe('Fragment', () => {
 		expect(scratch.textContent).to.equal('21');
 	});
 
+	it('should handle changing node type within a Component that returns a Fragment #1326', () => {
+		class X extends Component {
+			render() {
+				return this.props.children;
+			}
+		}
+
+		/** @type {(newState: any) => void} */
+		let setState;
+		class App extends Component {
+			constructor(props, context) {
+				super(props, context);
+
+				this.state = { i: 0 };
+				setState = this.setState.bind(this);
+			}
+
+			render() {
+				if (this.state.i === 0) {
+					return (
+						<div>
+							<X>
+								<span>1</span>
+							</X>
+							<X>
+								<span>2</span>
+								<span>2</span>
+							</X>
+						</div>
+					);
+				}
+
+				return (
+					<div>
+						<X>
+							<div>1</div>
+						</X>
+						<X>
+							<span>2</span>
+							<span>2</span>
+						</X>
+					</div>
+				);
+			}
+		}
+
+		render(<App />, scratch);
+		expect(scratch.innerHTML).to.equal(div([
+			span(1),
+			span(2),
+			span(2)
+		].join('')));
+
+		setState({ i: 1 });
+
+		clearLog();
+		rerender();
+
+		expect(scratch.innerHTML).to.equal(div([
+			div(1),
+			span(2),
+			span(2)
+		].join('')));
+		expectDomLogToBe([
+			'<span>1.remove()',
+			// TODO: Why is it re-appending all the children 😢
+			'<div>.appendChild(#text)',
```

Comment:
Somehow `oldDom` (previously `childDom`) is `null` here. On top of that the index of the old vnode and the new vnode are the same, but when I add a check for that the resulting dom will be incorrect :thinking:

Reply:
Interesting... Yeah, need to figure out why it goes to null

Yup, that was my plan. I'll come back to this after this PR goes through.